backend/src/main/java/fk/prof/backend/model/election/LeaderDiscoveryStore.java

```diff
@@ -0,0 +1,24 @@
+package fk.prof.backend.model.election;
+
+public interface LeaderDiscoveryStore {
+  /**
+   * Accepts null argument as well, which is treated as removing leader mapping from the underlying store
+   *
+   * @param ipAddress
+   */
+  void setLeaderIPAddress(String ipAddress);
```

Comment:
Separated get and set functionality in two interfaces. InMemoryStore implements both